### PR TITLE
NIFI-3251: Updating authorization requirements for removing components

### PR DESCRIPTION
**Pull request by mcgilman**
- Requiring WRITE permissions to the parent resource when attempting to remove a component.
- Updating expired certificates in the REST API integration tests.

NOTE: There are a couple places (templates and access policies) where the UI does not know if the user will have all required permissions ahead of time. I updated the UI where these details are known but think it makes sense to add these changes and update the remaining places one those details are available. The behavior now will simply provide a dialog indicating that the user does not have permissions to perform the desired action.

### CODE REVIEW

**Comment by jvwing** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-services.js [hunk 1]
```diff
@@ -497,6 +497,36 @@ nf.ControllerServices = (function () {
     };
 
     /**
+     * Determines if the user is authorized for the parent of the specified controller service.
+     *
+     * @param dataContext
+     * @returns {boolean}
+     */
+    var isAuthorizedForControllerServiceParent = function (dataContext) {
```
Comment (jvwing) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-services.js]: The method naming seems imprecise here, how about `canWriteControllerServiceParent` or something that clarifies which permission or action is authorized?

Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-services.js]: Agreed. Will update.

**Comment by jvwing** [comment]
@mcgilman Did you consider modifying the `isDeletable` method in nf-canvas-utils to check process group write access?  This is currently enabling the Delete button in the operate palette when the user has write permissions on a processor but not on a process group.

**Comment by mcgilman** [comment]
Yep, I'm tracking. Looks like that will provide more consistent coverage when checking if that behavior should be available.

**Comment by mcgilman** [comment]
I've updated the PR with the changes suggested thus far. Thanks again @jvwing!

**Comment by jvwing** [comment]
@mcgilman I think this looks good, it worked well in my testing. The integration tests are passing again.  These changes didn't break any tests, but none of the tests in nifi-web-api really exercise them, and the test framework does not make it easy to add something for this.  At the risk of being a bad committer, I merged anyways.

Thanks for contributing this.

**Pull request closed without merging**